corehq/apps/ota/utils.py

```diff
@@ -0,0 +1,58 @@
+from casexml.apps.case.xml import V2
+from casexml.apps.phone.restore import RestoreConfig, RestoreParams
+from corehq.apps.domain.models import Domain
+
+from .models import DemoUserRestore
+
+
+def turn_off_demo_mode(commcare_user):
+    """
+    Turns demo mode OFF for commcare_user and deletes existing demo restore
+    """
+    commcare_user.is_demo_user = False
+
+    # delete old restore
+    old_restore_id = commcare_user.demo_restore_id
+    if old_restore_id:
+        old_restore = DemoUserRestore.objects.get(uuid=old_restore_id)
+        old_restore.delete()
+
+    commcare_user.demo_restore_id = None
+    commcare_user.save()
+
+
+def turn_on_demo_mode(commcare_user, domain):
+    """
+    Turns demo mode ON for commcare_user, and resets restore to latest
+    """
+    # ToDo This should be in a task
+    reset_demo_user_restore(commcare_user, domain)
+    commcare_user.is_demo_user = True
+    commcare_user.save()
+
+
+def reset_demo_user_restore(commcare_user, domain):
+    """
+    Updates demo restore for the demo commcare_user
+    """
+    assert commcare_user.domain == domain
+
+    # get latest restore
+    restore = RestoreConfig(
+        project=Domain.get_by_name(domain),
+        user=commcare_user.to_casexml_user(),
+        params=RestoreParams(version=V2),
+    ).get_payload().as_string()
```

Comment:
I think it's worth it otherwise the entire response gets loaded in to
memory. I think the return object can either be a `CachedResponse` or
a `FileRestoreResponse`.
It shouldn't be too hard to implement.